src/views/Leaderboard.vue

```diff
@@ -0,0 +1,202 @@
+            ).toFixed(2)
+          }}%
+        </td>
+      </tr>
+    </table>
+    <div
+      style="
+        width: 100%;
+        display: flex;
+        justify-content: flex-end;
+        padding-top: 10px;
+      "
+    >
+      <button class="btn btn-primary" @click="onFirst" style="margin: 5px">
+        First
+      </button>
+      <button class="btn btn-primary" @click="onPrevious" style="margin: 5px">
+        {{ "<" }}
+      </button>
+      <button
+        class="btn btn-primary"
+        @click="onSwitchElementsToDisplay"
+        style="margin: 5px;"
+      >
+        {{ `${currentIndex + 1} - ${currentIndex + elementsToDisplay}` }}
+      </button>
+      <button class="btn btn-primary" @click="onNext" style="margin: 5px">
+        {{ ">" }}
+      </button>
+      <button class="btn btn-primary" @click="onLast" style="margin: 5px">
+        Last
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+import holder from "../store/data/holder.json";
+
+const latestBlock = 14611087;
+
+export default {
+  name: "Leaderboard",
+  data() {
+    return {
+      leaderboard: holder,
+      elementsToDisplay: 10,
+      currentIndex: 0,
+    };
+  },
+  async mounted() {
+    this.fetchTransfers(latestBlock);
+  },
+  methods: {
+    onFirst() {
+      this.currentIndex = 0;
+    },
+    onPrevious() {
+      const newIndex = this.currentIndex - this.elementsToDisplay;
+      if (newIndex < 0) {
+        this.onFirst();
+      } else {
+        this.currentIndex = newIndex;
+      }
+    },
+    onSwitchElementsToDisplay() {
+      switch (this.elementsToDisplay) {
+        case 10:
+          if (this.currentIndex + 25 > this.leaderboard.length) this.currentIndex -= 15;
+          this.elementsToDisplay = 25;
+          break;
+        case 25:
+          if (this.currentIndex + 50 > this.leaderboard.length) this.currentIndex -= 25;
+          this.elementsToDisplay = 50;
+          break;
+        case 50:
+          if (this.currentIndex + 100 > this.leaderboard.length) this.currentIndex -= 50;
+          this.elementsToDisplay = 100;
+          break;
+        default:
+          this.elementsToDisplay = 10;
+          break;
+      }
+    },
+    onNext() {
+      const newIndex = this.currentIndex + this.elementsToDisplay;
+      if (this.leaderboard.length - this.elementsToDisplay < newIndex) {
+        this.onLast();
+      } else {
+        this.currentIndex = newIndex;
+      }
+    },
+    onLast() {
+      this.currentIndex = this.leaderboard.length - this.elementsToDisplay;
+    },
+    async fetchTransfers(startBlock) {
+      const address = "0x93a796b1e846567fe3577af7b7bb89f71680173a";
+      const topic = "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef";
+      const apikey = "REPLACE_WITH_YOUR_API_KEY";
+      const url = `https://api.etherscan.io/api?module=logs&action=getLogs&fromBlock=${startBlock}&toBlock=latest&address=${address}&topic0=${topic}&apikey=${apikey}`;
```

Comment:
can we not use the code that already exists in the Store? this already exists and doesnt require a stupid api key <.<

Reply:
yes. i reworked it. have a look if you have time